include/oneapi/dpl/experimental/kt/esimd_radix_sort.h

```diff
@@ -0,0 +1,84 @@
+// -*- C++ -*-
+//===-- esimd_radix_sort.h --------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------===//
+
+#ifndef _ONEDPL_KT_ESIMD_RADIX_SORT_H
+#define _ONEDPL_KT_ESIMD_RADIX_SORT_H
+
+#include "kernel_param.h"
+#include "../../pstl/utils_ranges.h"
+#include "../../pstl/hetero/dpcpp/utils_ranges_sycl.h"
+#include "../../pstl/hetero/dpcpp/parallel_backend_sycl_utils.h"
+
+#include <cstdint>
+
+#include "internal/esimd_radix_sort_utils.h"
+#include "internal/esimd_radix_sort_dispatchers.h"
+
+namespace oneapi::dpl::experimental::kt::esimd
+{
+
+// TODO: make sure to provide sufficient diagnostic if input does not allow either reading or writing
+template <bool __is_ascending = true, ::std::uint8_t __radix_bits = 8, typename _KernelParam, typename _KeysRng>
+sycl::event
+radix_sort(sycl::queue __q, _KeysRng&& __keys_rng, _KernelParam __param = {})
```

Comment:
May be better to separate declaration of public API from implementation?

Reply:
It has benefits and drawbacks:
 - (+) It is easier to reference the API here as documentation.
 - (-) Duplication.

I am inclined to leave it as is unless there are more benefits.

Reply:
Unless I am mistaken, I think it breaks convention with the rest of the library.  As far as I know, we have forward definitions of the public APIs throughout oneDPL. Of course, with KT, we can perhaps make a change in this decision.

Reply:
As far as I understand, other APIs require inclusion of two headers: `execution` for the policies and one among `algorithm` `memory`, `numeric`  and other headers for specific algorithms. Forward declarations are used to support different inclusion orders (when `execution` is included before or after e.g.  `algorithm`).

I am not sure if such a convention exists, because it looks to me as a solution of a technical problem which does not exist in KT. 

@MikeDvorskiy, Is there such a convention? Would it be better to have forward declarations in the KT?

Reply:
oneDPL has one rule regarding "separate declaration of public API" for "standard" parallel algorithm API, located into algorithm, numeric, memory. A reason and logic of that is described here:
https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/numeric#L26
Lines  26-35